libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -698,7 +698,13 @@ IotNetworkError_t IotNetworkAfr_Destroy( void * pConnection )
         if( pNetworkConnection->receiveCallback != NULL )
         {
             EventBits_t connectionFlags;
-            connectionFlags = xEventGroupGetBits( ( EventGroupHandle_t ) &( pNetworkConnection->connectionFlags ) );
+
+            /* Clear the flag indicating the receive task has exited. */
+            connectionFlags = xEventGroupWaitBits( ( EventGroupHandle_t ) &( pNetworkConnection->connectionFlags ),
```

Comment:
The bit should be cleared after the assert check of the `_FLAG_RECEIVE_TASK_EXITED` flag

Reply:
Removed the wait